tools/infrasys/src/keys.rs

```diff
@@ -89,9 +87,13 @@ impl KMSKeyConfigExt for KMSKeyConfig {
                     missing: "key_alias",
                 })?
             );
-            let cfn_client = CloudFormationClient::new(
-                Region::from_str(region).context(error::ParseRegionSnafu { what: region })?,
-            );
+
+            let config = aws_config::from_env()
```

Comment:
This is where I'd be the most worried and concerned.  In what order and precedence does the new SDK gather config/credentials compared to the old one?

Reply:
My understanding is the default values for things that did not have an override are inserted at the `.load()` stage. If it's something that was overridden, but invalid (for example setting the region to a non-existent one) then the region will be left blank rather than receiving a default value.